compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -1412,16 +1413,45 @@ fn start_executing_work<B: ExtraBackendMethods>(
             // re-acquire a token to continue running more work. We may also not
             // actually drop a token here if the worker was running with an
             // "ephemeral token"
-            let mut free_worker = |worker_id| {
-                if main_thread_worker_state == MainThreadWorkerState::LLVMing {
-                    main_thread_worker_state = MainThreadWorkerState::Idle;
+            let free_worker = |worker_id,
+                               running: &mut _,
+                               main_thread_worker_state: &mut _,
+                               free_worker_ids: &mut Vec<_>| {
```

Comment:
This awkward change is because we can't have two closures over the same mutable state, even if they won't be used at the same time :(